lib/src/snapd_client.dart

```diff
@@ -1418,6 +1475,21 @@ class SnapdClient {
     return snaps;
   }
 
+  /// List all assertions of the given [assertion] type.
+  /// If no [assertion] type is provided, lists available assertion types.
+  Future<Map<String, dynamic>> assertions(
+      {String? assertion, Map<String, String>? params}) async {
```

Comment:
Instead of a generic params this should be a list of known params, e.g. `{String? series, bool remote = false, String ?snapId}` and then this method makes the map from these.

Reply:
Ah, I wasn't aware of that. You should document what [params] is and either some of the possible params or link to some documentation about them.

Reply:
And consider if it makes sense to add a helper method which does have the params set, e.g. `getAccountAssertions()`

Reply:
There are [quite a few assertion types](https://snapcraft.io/docs/snapd-rest-api#heading--assertions), but that could be theoretically possible.

Reply:
Up to you if you think it would be worthwhile - just trying to reduce the changes of an app using this library from making a mistake and having to read a lot of documentation.